### ISSUE TITLE
RFE: Retry fetching of trust anchor for DNSSEC validation

### ISSUE THREAD

**Issue by jonathanunderwood** [getdnsapi/getdns#408]
Currently stubby tries to retrieve a trust anchor on startup. If that fails, stubby gives up on DNSSEC validation. This can be problematic if stubby is started before the wan connection is fully up. In OpenWRT we work around this by using triggers. But it would b better if stubby would try to retrieve the trust anchor ahead of subsequent lookups, otherwise stubby will be silently not validating, and the user could be misled as to the validity of the returned records.

This was discussed [here](https://github.com/openwrt/packages/pull/7122#issuecomment-425899111), and @saradickinson recommended raising an issue here for the attention of @wtoorop .

**Comment by jonathanunderwood** [comment]
I'm sure this thinking is faulty, but I'll ask anyway: If I've configured stubby to require dnssec validation, wouldn't I expect to get SERVFAIL until the trust anchor is in place? IOW, to me, it seems surprising to ever get non-validated responses from stubby if I've configured stubby to validate. What am I missing?

**Comment by saradickinson** [comment]
The DNSSEC configuration is quite subtle and I can see why this leads to confusion fo users. None of the current settings do what you want (hard fail if I DNSSEC is configured but stubby can't fetch the trust anchor - I've talked to @wtoorop and agreed we need a new option for this and much better logging of the failure mode). 

Have a read of this section of the API: 
https://getdnsapi.net/documentation/spec/#31-extensions-for-dnssec

Stubby only SERVFAILS if the answer is BOGUS, without a trust anchor I believe everything comes back INDETERMINATE....

**Comment by jonathanunderwood** [comment]
As far as I can see, the backoff time isn't exposed via the Stubby configuration, is it?

**Comment by wtoorop** [comment]
It should be, because stubby uses the [`getdns_context_config()`](https://getdnsapi.net/functions/getdns_context_config.html) function under the hood.

**Comment by wtoorop** [comment]
Just tested it and it works.  I updated the example configuration file with an example: getdnsapi/stubby@006e43fd .

**Comment by jonathanunderwood** [comment]
Great, thanks. I guess the same thing applies to the new TLS options as well?

**Comment by wtoorop** [comment]
Yes, I'll add those to the example config file as well (as you requested in getdnsapi/stubby#148 )
